packages/browser/__tests__/Session.spec.ts

```diff
@@ -485,6 +497,81 @@ describe("Session", () => {
         redirectUrl: "https://some.redirect/url",
       });
     });
+
+    it("does nothing if the developer has not enabled session restore", async () => {
+      const sessionId = "mySession";
+      mockLocalStorage({
+        [KEY_CURRENT_SESSION]: sessionId,
+      });
+      mockLocation("https://mock.current/location");
+      const mockedStorage = new StorageUtility(
+        mockStorage({
+          [`${USER_SESSION_PREFIX}:${sessionId}`]: {
+            isLoggedIn: "true",
+          },
+        }),
+        mockStorage({})
+      );
+      const clientAuthentication = mockClientAuthentication({
+        sessionInfoManager: mockSessionInfoManager(mockedStorage),
+      });
+      clientAuthentication.validateCurrentSession = jest
+        .fn()
+        .mockResolvedValue({
+          issuer: "https://some.issuer",
+          clientAppId: "some client ID",
+          clientAppSecret: "some client secret",
+          redirectUrl: "https://some.redirect/url",
+        });
+      clientAuthentication.handleIncomingRedirect = jest
+        .fn()
+        .mockResolvedValue(undefined);
+      clientAuthentication.login = jest.fn();
+
+      const mySession = new Session({ clientAuthentication });
+      await mySession.handleIncomingRedirect("https://some.redirect/url");
+
+      expect(window.localStorage.getItem(KEY_CURRENT_URL)).toBeNull();
+    });
+
+    it("does nothing if the developer has disabled session restore", async () => {
+      const sessionId = "mySession";
+      mockLocalStorage({
+        [KEY_CURRENT_SESSION]: sessionId,
+      });
+      mockLocation("https://mock.current/location");
+      const mockedStorage = new StorageUtility(
+        mockStorage({
+          [`${USER_SESSION_PREFIX}:${sessionId}`]: {
+            isLoggedIn: "true",
+          },
+        }),
+        mockStorage({})
+      );
+      const clientAuthentication = mockClientAuthentication({
+        sessionInfoManager: mockSessionInfoManager(mockedStorage),
+      });
+      clientAuthentication.validateCurrentSession = jest
+        .fn()
+        .mockResolvedValue({
+          issuer: "https://some.issuer",
+          clientAppId: "some client ID",
+          clientAppSecret: "some client secret",
+          redirectUrl: "https://some.redirect/url",
+        });
+      clientAuthentication.handleIncomingRedirect = jest
+        .fn()
+        .mockResolvedValue(undefined);
+      clientAuthentication.login = jest.fn();
+
+      const mySession = new Session({ clientAuthentication });
+      await mySession.handleIncomingRedirect({
+        url: "https://some.redirect/url",
+        restorePreviousSession: false,
+      });
+
+      expect(window.localStorage.getItem(KEY_CURRENT_URL)).toBeNull();
```

Comment:
```suggestion
      expect(window.localStorage.getItem(KEY_CURRENT_URL)).toBeNull();
      expect(clientAuthentication.login).not.toHaveBeenCalled();
```
I think adding this to the test would verify more striclty if the silent authentication has not been triggered.

Reply:
Actually that was my initial plan, but then I saw that other "it does nothing" tests added this check :) 4d5b2132